src/engine/strat_engine/backstore/util.rs

```diff
@@ -36,3 +39,44 @@ pub fn hw_lookup(dev_node_search: &Path) -> EngineResult<Option<String>> {
 
     result
 }
+
+// Probes a block device using the libblkid library and returns Some(None) if the device is empty or
+// Some(String) with the String having similar content as the output of `blkid -p <device>.
+pub fn device_signature(path: &Path) -> Result<Option<String>, blkid::BlkidError> {
```

Comment:
Given that you've added a new class of error to errors.rs, this function should be able to return an ```EngineResult```, by implicitly promoting BlkidErrors.

Reply:
Yes, good point.